oddt/surface.py

```diff
@@ -0,0 +1,166 @@
+from scipy.spatial.distance import cdist
+
+
+def generate_surface_marching_cubes(molecule, remove_hoh=False, scaling=1, probe_radius=1.4):
+    """Generates a molecular surface mesh using the marching_cubes_lewiner method from scikit-image.
+
+    Parameters
+    ----------
+    molecule : oddt.toolkit.Molecule object
+        Molecule for which the surface will be generated
+    remove_hoh : bool (default = False)
+        If True, remove waters from the molecule before generating the surface
+    scaling : float (default = 1)
+        Expands the grid in which computation is done by a factor of scaling.
+        Results in a more accurate representation of the surface, but increases
+        computation time.
+    probe_radius : float (default = 1.4)
+        Radius of a ball used to patch up holes inside the molecule resulting from
+        some molecular distances being larger (usually in protein). Basically reduces
+        the surface to one accesible by other molecules of radius smaller than probe_radius.
+
+    Returns
+    -------
+    verts : numpy array
+        Spatial coordinates for mesh vertices.
+    faces : numpy array
+        Faces are defined by referencing vertices from verts.
+    """
+    # Copy and pre-process the molecule
+    mol = molecule.clone
+    mol.removeh()
+    if remove_hoh:
+        mol.protein = True
+        no_hoh = mol.atom_dict['resname'] != 'HOH'
+        atom_dict = mol.atom_dict[no_hoh]
+    else:
+        atom_dict = mol.atom_dict
+    # Take a molecule's coordinates and atom radii and scale if necessary
+    coords = atom_dict['coords'] * scaling
+    radii = atom_dict['radius'] * scaling
+    # Create a ball for each atom in the molecule
+    ellip_list = [ball(radius, dtype=bool) for radius in radii]
+    ellip_radii = np.array([ellip.shape[0] for ellip in ellip_list])
+    # Transform the coordinates because the grid starts at (0, 0 ,0)
+    min_coords = np.min(coords, axis=0)
+    max_rad = np.max(ellip_radii, axis=0)
+    adjusted = np.round(coords - min_coords + max_rad * 5).astype(np.int64)
+    offset = adjusted[0] - coords[0]
+    # Calculate boundries in the grid for each ball.
+    ellip_coord_min = (adjusted.T - np.floor(ellip_radii / 2).astype(np.int64)).T
+    ellip_coord_max = (ellip_coord_min.T + ellip_radii).T
+    # Create the grid
+    grid = np.zeros(shape=ellip_coord_max.max(axis=0) + 8 * scaling, dtype=bool)
+    # Place balls in grid
+    for ellip, coord_min, coord_max in zip(ellip_list, ellip_coord_min, ellip_coord_max):
+        grid[coord_min[0]:coord_max[0],
+             coord_min[1]:coord_max[1],
+             coord_min[2]:coord_max[2]] += ellip
+    spacing = (1 / scaling,) * 3
+    # Hole-filling with morphological closing
+    grid = binary_closing(grid, ball(probe_radius * 2 * scaling))
+    # Marching cubes
+    verts, faces, normals, values = measure.marching_cubes_lewiner(grid, level=0, step_size=1, spacing=spacing)
+    return verts - offset / scaling, faces
```

Comment:
I think this part requires a comment - at first I thought there are missing parentheses around `verts - offset`. Point out that `verts` are already in the orignal scale.